### ISSUE TITLE
Nodes can be built on some unsuitable terrain, resulting in their immediate destruction when building is complete

### ISSUE THREAD

**Issue by VReaperV** [Unvanquished/Unvanquished#2637]
There're certain places where a node can be built, but once the building is completed the node gets destroyed. Note that this does not happen with other buildings, including alien ones.

So far, this is the only place I've seen this issue happen (map `chasm`):
![unvanquished_2023-04-13_140716_000](https://user-images.githubusercontent.com/10687142/231741621-9af4bf16-55e3-46f9-9589-0a7e7efb8c07.jpg)

Nodes being destroyed in this place may be a map bug, but it shouldn't be possible to build them there in the first place.

**Comment by sweet235** [comment]
You could set `cg_drawPosition` to 1 for making screenshots like this one. It helps a little in reproducing.

On topic: sometimes, new alien buildings get destroyed if an egg's spawn area would be blocked by completing the construction of the new building. I wonder if this is related. Did you ever see an egg get destroyed by map geometry alone?

**Comment by VReaperV** [comment]
> You could set `cg_drawPosition` to 1 for making screenshots like this one. It helps a little in reproducing.
> 
> On topic: sometimes, new alien buildings get destroyed if an egg's spawn area would be blocked by completing the construction of the new building. I wonder if this is related. Did you ever see an egg get destroyed by map geometry alone?

Ah, thanks. The node is around -1607 -568 75 on this screenshot, however a lot of the slpoed area nearby shows this problem.

I don't think I've seen an alien egg get destroyed by map geometry. I do think it could be related. I assume this is for the most part done automatically, perhaps in this case the irregular sloped geometry is making it problematic. Regardless though, it shouldn;'t be possible to build a node on such a location.

I think the reason an egg doesn't get destroyed here is because dretches and grangers have a lot smaller hitboxes than humans, so perhaps there's some sort of volume that prevents spawning above that intersects with the human hitbox, but not a dretch/granger one.

**Comment by sweet235** [comment]
> Regardless though, it shouldn;'t be possible to build a node on such a location.

This is certainly a bug. But I'm not sure if this is easy to fix. Might be a complicated client/server thing.

> so perhaps there's some sort of volume that prevents spawning above that intersects with the human hitbox, but not a dretch/granger one

I thought so too. But then i noticed another curious thing: when building alien buildings on that slope, the creep texture on the floor is not drawn. Maybe this is a map bug, or some unfortunate combination of bugs.

One more remotely related observation: the Overmind is the only alien building that can win against an egg in the battle for spawn space. I've known for some time that constructing the Overmind can result in destruction of the last spawn. But a few days ago, I observed this in an actual online game, on my team. Now I wonder if there can be a similar fight between a telenode and the reactor.

**Comment by VReaperV** [comment]
> I thought so too. But then i noticed another curious thing: when building alien buildings on that slope, the creep texture on the floor is not drawn. Maybe this is a map bug, or some unfortunate combination of bugs.

I think this has to do with map geometry. Here's another place in chasm where this happens:
![unvanquished_2023-04-13_164734_000](https://user-images.githubusercontent.com/10687142/231780217-9d1605ee-a277-4809-a183-dd61b77e8553.jpg)
And here's one from the map `ancient-remains`:
![unvanquished_2023-04-13_165100_000](https://user-images.githubusercontent.com/10687142/231780330-a7fac868-044c-497c-83c8-ea562927b361.jpg)
Building a node just a little bit further works fine however:
![unvanquished_2023-04-13_165110_000](https://user-images.githubusercontent.com/10687142/231780527-3e21d565-0656-46f2-9edc-324ae854a973.jpg)
So I'm not sure if the idea about a volume intersecting the hitbox is true.

> Now I wonder if there can be a similar fight between a telenode and the reactor.

That is possible, yes:
![unvanquished_2023-04-13_164214_000](https://user-images.githubusercontent.com/10687142/231780976-17cb1490-baad-4ec3-b565-98fc04f2874c.jpg)

**Comment by sweet235** [comment]
This is the best issue I read in some time.

**Comment by VReaperV** [comment]
> I have to ask: do we really want the main buildable to win against the last spawn? I don't know. A difficult question, maybe.

I'd say it goes against the reason for having a restriction on deconstructing the last spawn. I can also see a situation where a team has to build their main structure on that specific spot but can't afford to move the spawn, although it would be really rare.

**Comment by bmorel** [comment]
> I have to ask: do we really want the main buildable to win against the last spawn? I don't know. A difficult question, maybe.

To me the answer is obvious: spawns are more important than RC/OM.
Spawns cost BPs, unlike the uniques. They also are the most important building in a team: with just a spawn, you can always rebuild the whole base, even if you're dead. With just a unique, well... you loose.

**Comment by VReaperV** [comment]
> To me the answer is obvious: spawns are more important than RC/OM.
> Spawns cost BPs, unlike the uniques. They also are the most important building in a team: with just a spawn, you can always rebuild the whole base, even if you're dead. With just a unique, well... you loose.

Suppose the human base is in a small room and all humans are spawned but the reactor is dead. The only place they can rebuild it is above the last node. If they can build rc, they can use the armory and medistation. If they only have a spawn and unpowered armory/medistation, then they quickly get slaughtered.

**Comment by bmorel** [comment]
yes, this is all very unlikely, except for aliens, actually. I think I never seen a telenode destroying anything, except when I tried it on purpose. Eggs killing other stuff is a rather common problem though.

**Comment by bmorel** [comment]
to check the problem is bbox related: try to build a miner, they're smaller.
Also, I tried to build an armoury there, and it stays around.

I do not know if that's related to BBox though: it might be because of a normal problem, as eggs can be built on walls and ceilings, unlike H stuff.

I agree building should not be allowed anyway for that kind of places.

**Comment by VReaperV** [comment]
> yes, this is all very unlikely, except for aliens, actually. I think I never seen a telenode destroying anything, except when I tried it on purpose. Eggs killing other stuff is a rather common problem though.

I see, I barely play aliens myself, so haven't run into that sort of situation. I think it should be either impossible to build there or communicated to the player in some way. Either way though, that might be out of scope of this issue.

**Comment by VReaperV** [comment]
> to check the problem is bbox related: try to build a miner, they're smaller. Also, I tried to build an armoury there, and it stays around.
> 
> I do not know if that's related to BBox though: it might be because of a normal problem, as eggs can be built on walls and ceilings, unlike H stuff.
> 
> I agree building should not be allowed anyway for that kind of places.

I meant the bbox of a human when they spawn on that telenode, sorry if the wording was confusing. I. e. a human has no room to spawn there (or so the game thinks), but for some reason doesn't stop the node's construction and instead just destroys it after it's been built.

**Comment by VReaperV** [comment]
> > to check the problem is bbox related: try to build a miner, they're smaller. Also, I tried to build an armoury there, and it stays around.
> > I do not know if that's related to BBox though: it might be because of a normal problem, as eggs can be built on walls and ceilings, unlike H stuff.
> > I agree building should not be allowed anyway for that kind of places.
> 
> I meant the bbox of a human when they spawn on that telenode, sorry if the wording was confusing. I. e. a human has no room to spawn there (or so the game thinks), but for some reason the game doesn't stop the node's construction and instead just destroys it after it's been built.
